### PR TITLE
✨New Style for the Modeler

### LINKED ISSUE #205: Invisible div catches mouseevents on workspace
#### Describe your issue

There are two divs(class names are `resize_div` and `toggle_div`).
They are invisible and contain buttons of the property panel.
Both divs are above workspace and catch mouseevents. In the screenshot you can't click on the `end process`-bpmn-element because of that.
Really a minor issue, but mildly irritating and easy to resolve.

screen shot 2018-03-15 at 14 14 52

#### What's your setup

Firefox. Problem should be the same for any setup.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### LINKED ISSUE #246: Implement new UX Theme
Implement the new UI/UX for BPMN-Studio.

### LINKED ISSUE #257: Electron: Allow toggling DevTools
#### Describe your issue

In production builds you are unable to toggle DevTools.

To allow better debugging we should allow this

#### How can others reproduce the issue?

Try to open DevTools via `CMD+OPTION+I`

#### Possible solution

Add hidden shortcut to open DevTools

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by Vyperus**
## What did you change?

This branch changes the look and feel of the "ProcessDef-details"-view (the modeler) drastically.
Overall it adds the new Css-Schemas while keeping the old functionalities.

Fixes #205, #259, #265   
Closes #246
Closes #257  

New Design:
bildschirmfoto 2018-04-25 um 15 40 08

New Design with open property-panel and solution-explorer:
bildschirmfoto 2018-04-25 um 15 40 26

## How can others test the changes?

You can test theese changes by opening a bpmn-diagram with bpmn-studio.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/modules/import-process-button/import-process-button.html [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
   <require from="./import-process-button.css"></require>
-  <require from="../choose-dialog/choose-dialog"></require>  
+  <require from="../choose-dialog/choose-dialog"></require>
 
   <input type="file" ref="fileInput" accept=".xml,.bpmn" files.bind="selectedFiles" class="hidden">
-  <a class="btn btn-primary" click.delegate="fileInput.click()">
-    <i class="glyphicon glyphicon-import"></i> Import Diagram
-  </a>
+  <button class="menu-bar__menu-center--action-button" click.delegate="fileInput.click()">
+      <i class="fas fa-upload"></i>
```
Comment (Paulomart) [inline src/modules/import-process-button/import-process-button.html]: Please fix indention

File: src/modules/processdef-detail/processdef-detail.scss [hunk 2]
```diff
@@ -127,3 +173,4 @@ bpmn-io {
   color: transparent;
 }
 
+
```
Comment (Paulomart) [inline src/modules/processdef-detail/processdef-detail.scss]: please remove extra newline

File: src/modules/processdef-detail/processdef-detail.html [hunk 3]
```diff
@@ -86,27 +125,55 @@ <h5 class="header">Process Instance
...
+      MAIN CANVAS
+    -->
+    <div class="window-layout__main js-main">
+      <bpmn-io show.bind="!showXMLView" view-model.ref="bpmn" xml.bind="process.xml"></bpmn-io>
+    </div>
+
+    <!--
+    STATUS BAR
+    -->
+    <div class="status-bar status-bar--system-macos">
```
Comment (Paulomart) [inline src/modules/processdef-detail/processdef-detail.html]: Was the statusbar before not showing on all views?

File: src/modules/processdef-detail/processdef-detail.ts [hunk 4]
```diff
@@ -153,19 +153,23 @@ export class ProcessDefDetail {
...
-    this.processEngineService.getProcessDefById(this.processId)
+  private refreshProcess(): Promise<IProcessDefEntity> {
+    return this.processEngineService.getProcessDefById(this.processId)
       .then((result: any) => {
         if (result && !result.error) {
           this._process = result;
+          return this._process;
         } else {
           this._process = null;
+          return result.error;
```
Comment (Paulomart) [inline src/modules/processdef-detail/processdef-detail.ts]: you should consider `rejecting` the promise here

File: src/modules/processdef-detail/processdef-detail.ts [hunk 5]
```diff
@@ -197,22 +201,23 @@ export class ProcessDefDetail {
   }
 
   public async saveDiagram(): Promise<void> {
+    if (!this.saveButton.disabled) {
```
Comment (Paulomart) [inline src/modules/processdef-detail/processdef-detail.ts]: Please use eaily exits here: 

```javascript
if (this.saveButton.disabled) {
  return;
}
```

File: src/modules/processdef-detail/processdef-detail.html (hunk 3 above)
Comment (Vyperus) [reply src/modules/processdef-detail/processdef-detail.html]: There was no "statusbar" before

File: src/modules/notification/notification.service.ts [hunk 6]
```diff
@@ -1,3 +1,4 @@
+import {Toastr} from 'toastr';
```
Comment (alexanderkasten) [inline src/modules/notification/notification.service.ts]: This line can and should get removed if the @types/toastr package is installed.

**Comment by LeoTT** [comment]
Fixes #205 

1. cannot differentiate between disabled and enabled nav buttons

screen shot 2018-04-24 at 16 42 11

Design, publish and dashboard look the same, but from my current context only dashboard is clickable.
The disabled buttons have no hovereffect, giving me the illusion that they are no buttons.
You could 
- not show disabled buttons at all (different contexts = different navbar)
- make different styles for `not selected` and `disabled`
- add `disable`-hover effect (better yet additional tooltip why it's disabled)


2. the left border of the property panel(for bpmn elements in the details tab) has a hover effect which makes it look like it's resizable (when it's not).

3. the xml-editor does not allow me to scroll to the bottom
screen shot 2018-04-24 at 16 53 18

4. state of the process solution explorer changes when I use navigation

Going to the dashboard makes the explorer appear.

5. hover effect of the folder icon of process solution explorer should be blue

Nitpicky, but that's how it goes for other nav elements.

6. Some navigation elements are not reachable with tab

I don't know the priority of accessibility, but it's good practice and not much work.
a-tags with no href attribute do not react to tab, you could use buttons instead.

File: src/modules/user-login/user-login.html [hunk 7]
```diff
@@ -1,31 +1,26 @@
...
-    </li>
-  </ul>
+  <div class="btn-group user-login">
+    <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+      <span>Hi, ${identity ? identity.name : 'Guest'} </span>
+    </a>
+    <img class="user-profile-image" src="src/resources/images/default-avatar.png">
+    <div class="dropdown-menu dropdown-menu-right user-profile-dropdown">
+      <div class="user-profile-dropdown-content clearfix">
+          <form if.bind="!isLoggedIn" submit.delegate="login()" role="form">
```
Comment (alexanderkasten) [inline src/modules/user-login/user-login.html]: wrong indentation

**Pull request merged**